router.py

```diff
@@ -44,8 +56,43 @@ def receive_packet(self, packet):
             packet (Packet): The packet we received.
 
         """
-        pass
 
-    def make_routing_table(self):
-        """Makes the routing table based on the links."""
-        pass
\ No newline at end of file
+        # If we get a routing packet, we need to update the routing table
+        if isinstance(packet, RoutingPacket):
+            self.update_routing_table(packet, link_id)
+        else: # We need to send the packet to its dest host
+            self.send_packet(packet)
```

Comment:
Is this right?  Is self the source?  (it probs is, I'm just confused cause the flow life is the only one I know)